src/core/intent_lock.h

```diff
@@ -3,6 +3,8 @@
 //
 #include <assert.h>
 
+#include <ostream>
```

Comment:
do you really need it?
Or maybe you can use iosfwd

Reply:
I **define**
```
friend std::ostream& operator<<(std::ostream& o, const IntentLock& lock) {
```
at line 63. you tell me if i need it.

Reply:
ok I see. Please feel free to ignore next, but as I understand it's for debug/log purposes, ostream is a really huge header maybe we can move include into cpp or make a forward declaration of this operator and add implementation in another place.

Reply:
it can be done but I do not want to increase the scope of this PR. I just added a missing header that is already present somewhere, otherwise it would not compile.